src/Tools/Common/Commands/Utils.cs

```diff
@@ -59,66 +69,66 @@ public static bool ValidateArgumentsForChildProcess(int processId, string name,
-                Console.WriteLine("Must specify either --process-id, --name, or --diagnostic-port.");
+                Console.WriteLine("Must specify either --process-id, --name, --diagnostic-port, or --dsrouter.");
                 return false;
             }
             else if (processId < 0)
             {
                 Console.WriteLine($"{processId} is not a valid process ID");
                 return false;
             }
-            else if (processId != 0 && !string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(port))
-            {
-                Console.WriteLine("Only one of the --name, --process-id, or --diagnostic-port options may be specified.");
-                return false;
-            }
-            else if (processId != 0 && !string.IsNullOrEmpty(name))
-            {
-                Console.WriteLine("Only one of the --name or --process-id options may be specified.");
-                return false;
-            }
-            else if (processId != 0 && !string.IsNullOrEmpty(port))
-            {
-                Console.WriteLine("Only one of the --process-id or --diagnostic-port options may be specified.");
-                return false;
-            }
-            else if (!string.IsNullOrEmpty(name) && !string.IsNullOrEmpty(port))
+            else if ( processId != 0 && (!string.IsNullOrEmpty(name) || !string.IsNullOrEmpty(port) || !string.IsNullOrEmpty(dsrouter))
+                    || !string.IsNullOrEmpty(name) && (!string.IsNullOrEmpty(port) || !string.IsNullOrEmpty(dsrouter))
+                    || !string.IsNullOrEmpty(port) && !string.IsNullOrEmpty(dsrouter))
             {
-                Console.WriteLine("Only one of the --name or --diagnostic-port options may be specified.");
+                Console.WriteLine("Only one of the --name, --process-id, --diagnostic-port, or --dsrouter options may be specified.");
                 return false;
             }
-            // If we got this far it means only one of --name/--diagnostic-port/--process-id was specified
+            // If we got this far it means only one of --name/--diagnostic-port/--process-id/--dsrouter was specified
             else if (!string.IsNullOrEmpty(port))
             {
                 return true;
             }
             // Resolve name option
             else if (!string.IsNullOrEmpty(name))
             {
-                processId = CommandUtils.FindProcessIdWithName(name);
-                if (processId < 0)
+                if ((processId = FindProcessIdWithName(name)) < 0)
                 {
                     return false;
                 }
             }
-            else if (processId == 0)
+            else if (!string.IsNullOrEmpty(dsrouter))
             {
-                Console.WriteLine("One of the --name, --process-id, or --diagnostic-port options must be specified when attaching to a process.");
-                return false;
+                ConsoleColor currentColor = Console.ForegroundColor;
```

Comment:
Maybe all this should go into a FindDSRouterProcess method or similar.

Reply:
ok for moving the colouring and message, but I think it make sense to keep the args validation in a single place